Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoft365Defender.yaml

```diff
@@ -27,7 +62,25 @@ ParserQuery: |
   ];
   // -- Common preprocessing to both input and outbound events
   let RawNetworkEvents =
-    DeviceNetworkEvents
+    DeviceNetworkEvents | where not(disabled)
+    // ***************  Prefilterring *****************************************************************
+    |where (isnull(starttime) or TimeGenerated>=starttime) and
+           (isnull(endtime) or TimeGenerated<=endtime) and
+           (isnull(dstportnumber) or dstportnumber == LocalPort or dstportnumber == RemotePort) and
+           (array_length(url_has_any)==0  or RemoteUrl has_any (url_has_any)) and
+           (array_length(srcipaddr_has_any_ipv4_prefix)==0 
```

Comment:
Wouldn't it make sense to just search for both source and dest lists in local and remote IPs in one (well two) passes instead of 4)?

Reply:
If I understand correctly you are suggesting something like this:

```
let unified_ip_prefixes=set_union(srcipaddr_has_prefix,dstipaddr_has_prefix);
XXXXX
| where array_length(ip_prefixes==0) or
     has_any_ipv4_prefix(RemoteIP, unified_ip_prefixes) or has_any_ipv4_prefix(LocalIP, unified_ip_prefixes)

```​
I think it creates an OR logic (instead of AND) between parameters (srcipaddr_has_prefix, dstipaddr_has_prefix):
if none of dstipaddr_has_any matches (Local, Remote), but one of srcipaddr_has_any does
- we want it to return FALSE
- it will return TRUE
@oshezaf , WDYT?

Reply:
@oshezaf , not sure tagging above worked